pkgs/by-name/gi/gitbutler/package.nix

```diff
@@ -0,0 +1,177 @@
+    ]
+    ++ lib.optionals stdenv.isDarwin (
+      with darwin.apple_sdk.frameworks;
+      [
+        AppKit
+        CoreServices
+        Security
+        WebKit
+      ]
+    );
+
+  env = {
+    # `pnpm`'s `fetchDeps` and `configHook` uses a specific version of pnpm, not upstream's
+    COREPACK_ENABLE_STRICT = 0;
+
+    # we depend on nightly features
+    RUSTC_BOOTSTRAP = 1;
+
+    # we also need to have `tracing` support in `tokio` for `console-subscriber`
+    RUSTFLAGS = "--cfg tokio_unstable";
+
+    tauriBundle =
+      {
+        Linux = "deb";
+        Darwin = "app";
+      }
+      .${stdenv.hostPlatform.uname.system}
+        or (throw "No tauri bundle available for ${stdenv.hostPlatform.uname.system}");
+
+    ESBUILD_BINARY_PATH = lib.getExe (
+      esbuild.override {
+        buildGoModule =
+          args:
+          buildGoModule (
+            args
+            // rec {
+              version = "0.18.20";
+              src = fetchFromGitHub {
+                owner = "evanw";
+                repo = "esbuild";
+                rev = "v${version}";
+                hash = "sha256-mED3h+mY+4H465m02ewFK/BgA1i/PQ+ksUNxBlgpUoI=";
+              };
+
+              vendorHash = "sha256-+BfxCyg0KkDQpHt/wycy/8CTG6YBA/VJvJFhhzUnSiQ=";
+            }
+          );
+      }
+    );
+  };
+
+  buildPhase = ''
+    runHook preBuild
+
+    cargo tauri build --bundles "$tauriBundle"
+
+    runHook postBuild
+  '';
+
+  installPhase =
+    ''
+      runHook preInstall
+    ''
+    + lib.optionalString stdenv.isDarwin ''
+      mkdir -p $out/bin
+      cp -r target/release/bundle/macos $out/Applications
+      mv $out/Applications/GitButler.app/Contents/MacOS/GitButler $out/bin/git-butler
+      ln -s $out/bin/git-butler $out/Applications/GitButler.app/Contents/MacOS/GitButler
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      cp -r target/release/bundle/"$tauriBundle"/*/data/usr $out
+
+      desktop-file-edit \
+        --set-comment "A Git client for simultaneous branches on top of your existing workflow." \
+        --set-key="Keywords" --set-value="git;" \
+        --set-key="StartupWMClass" --set-value="GitButler" \
+        $out/share/applications/git-butler.desktop
+    ''
+    + ''
+      runHook postInstall
+    '';
+
+  # the `gitbutler-git` crate's checks do not support release mode
+  checkType = "debug";
+
+  nativeCheckInputs = [ git ];
+
+  passthru = {
+    updateScript = nix-update-script {
+      extraArgs = [
+        "--version-regex"
+        "release/(.*)"
+      ];
+    };
+  };
+
+  meta = {
+    description = "Git client for simultaneous branches on top of your existing workflow";
+    homepage = "https://gitbutler.com";
+    changelog = "https://github.com/gitbutlerapp/gitbutler/releases/tag/release/${version}";
+    mainProgram = "git-butler";
+    license = lib.licenses.fsl11Mit;
+    maintainers = with lib.maintainers; [ getchoo ];
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```

Comment:
```suggestion
    platforms = with lib.platforms; linux ++ darwin;
```

Reply:
the lack of `with lib.platforms;` here was intentional, as applying this diff results in `lib.platforms.darwin` being overshadowed by the top level `darwin` argument

![image](https://github.com/NixOS/nixpkgs/assets/48872998/b48fc7a4-cd76-4dce-a384-f9972f919a24)

Reply:
Oh, i didn't think of that, that's something new for me thx